tests/integration_tests/modules/test_kernel_modules.py

```diff
@@ -0,0 +1,120 @@
+"""Integration test for kernel_modules module."""
+import pytest
+from pycloudlib.lxd.instance import LXDInstance
+
+from cloudinit.subp import subp
+from tests.integration_tests.instances import IntegrationInstance
+
+ASCII_TEXT = "ASCII text"
+
+USER_DATA = """\
+#cloud-config
+packages:
+  - zfsutils-linux
+  - wireguard
+  - multipath-tools
+kernel_modules:
+  - name: lockd
+    load: true
+    persist:
+      options: "nlm_udpport=4045 nlm_tcpport=4045"
+  - name: wireguard
+  - name: zfs
+    load: true
+  - name: dm_multipath
+    persist:
+      blacklist: true
+"""
+
+KERNEL_MODULES_LXD = "lockd,zfs"
+
+
+def load_kernel_modules_lxd(instance: LXDInstance):
+    subp(
+        "lxc config set {} linux.kernel_modules {}".format(
+            instance.name, KERNEL_MODULES_LXD
+        ).split()
+    )
+
+
+@pytest.mark.user_data(USER_DATA)
+@pytest.mark.ci
+@pytest.mark.lxd_vm
+@pytest.mark.ubuntu
+class TestKernelModules:
```

Comment:
I think we could verify that there are no errors/warnings in the log. That will catch errors easily:

```suggestion
class TestKernelModules:
   def test_clean_log(self, class_client: IntegrationInstance):
       verify_clean_log(class_client)
```

Same for the other test class.